Content.Client/Guidebook/Controls/GuideReactionEmbed.xaml.cs

```diff
@@ -0,0 +1,143 @@
+    }
+
+    public bool TryParseTag(Dictionary<string, string> args, [NotNullWhen(true)] out Control? control)
+    {
+        control = null;
+        if (!args.TryGetValue("Reaction", out var id))
+        {
+            Logger.Error("Reaction embed tag is missing reaction prototype argument");
+            return false;
+        }
+
+        if (!_prototype.TryIndex<ReactionPrototype>(id, out var reaction))
+        {
+            Logger.Error($"Specified reaction prototype \"{id}\" is not a valid reaction prototype");
+            return false;
+        }
+
+        GenerateControl(reaction);
+
+        control = this;
+        return true;
+    }
+
+    private void GenerateControl(ReactionPrototype reactionPrototype)
+    {
+        var reactantMsg = new FormattedMessage();
+        var reactantsCount = reactionPrototype.Reactants.Count;
+        var i = 0;
+        foreach (var (product, reactant) in reactionPrototype.Reactants)
+        {
+            reactantMsg.AddMarkup(Loc.GetString(reactant.Catalyst ? "guidebook-catalyst-prototype-display" : "guidebook-reaction-prototype-display",
+                ("name", _prototype.Index<ReagentPrototype>(product).LocalizedName), ("ratio", reactant.Amount.ToString() + (reactionPrototype.Quantized ? "u" : ""))));
+            i++;
+            if (i < reactantsCount)
+                reactantMsg.PushNewline();
+        }
+        reactantMsg.Pop();
+        ReactantsLabel.SetMessage(reactantMsg);
+
+        if (reactionPrototype.MinimumTemperature > 0.0f)
+        {
+            MixLabel.Text = Loc.GetString("guidebook-reaction-mix-and-heat",
+                ("temperature", reactionPrototype.MinimumTemperature));
+        }
+
+        var productMsg = new FormattedMessage();
+        var productCount = reactionPrototype.Products.Count;
+        var u = 0;
+        foreach (var (product, ratio) in reactionPrototype.Products)
+        {
+            productMsg.AddMarkup(Loc.GetString("guidebook-reaction-prototype-display",
+                ("name", _prototype.Index<ReagentPrototype>(product).LocalizedName), ("ratio", ratio.ToString() + (reactionPrototype.Quantized ? "u" : ""))));
+            u++;
+            if (u < productCount)
+                productMsg.PushNewline();
+        }
+        if (_chemistryGuideData.ReactionGuideRegistry.TryGetValue(reactionPrototype.ID, out var guideEntryRegistry))
+        {
+            u = guideEntryRegistry.Count;
+            productMsg.PushNewline();
+            foreach (var (product, amount) in guideEntryRegistry)
+            {
+                productMsg.AddMarkup(Loc.GetString("guidebook-reaction-prototype-display",
+                    ("name", _prototype.Index<EntityPrototype>(product).Name), ("ratio", amount)));
+                u++;
+                if (u < productCount)
+                    productMsg.PushNewline();
+            }
+        }
+        productMsg.Pop();
+        ProductsLabel.SetMessage(productMsg);
+    }
```

Comment:
As above "u" is not a clear variable name and the code is unnecessarily hard to read.